src/Auth/UserIsAdminAuthValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace RebelCode\EddBookings\RestApi\Auth;
+
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Util\Normalization\NormalizeIterableCapableTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use Dhii\Validation\CreateValidationFailedExceptionCapableTrait;
+use Dhii\Validation\ValidatorInterface;
+
+/**
+ * An authorization validator that validates whether a user is an administrator.
+ *
+ * @since [*next-version*]
+ */
+class UserIsAdminAuthValidator implements ValidatorInterface
```

Comment:
Why not use [`AbstractValidatorBase`][]?

[`AbstractValidatorBase`]: https://github.com/Dhii/validation-base/blob/develop/src/AbstractValidatorBase.php

Reply:
Why _do_ use it? What's the benefit for such a simple validator implementation?

Reply:
There's some functionality in there, and you only need to implement the `_getValidationErrors()` method. But if you feel that this would un-necessarily complicate the validator, I have no problem with that.

Reply:
However, ask yourself this question, for example: What kind of exceptions does your `validate()` method throw, and does it comply with the standard interface?

Reply:
Especially since `AbstractBaseValidator` is tested.

Reply:
Well yeah, but what exactly does it test? A `try..catch` and exception creation? :laughing: Not really beneficial for a simple validator. Now that it's slightly less simple maybe it's a benefit, but the same amount of code is still left untested. Before it was called `validate()`, now it's called `_validate()` :laughing:

Reply:
Not `_getValidationErrors()`?

Also, the validation logic itself is, of course, untested in the abstract base class - coz it does not contain that logic. But the point is that all other logic, including guaranteed correct exceptions, is tested - so that the author only needs to test their validation logic.

Reply:
Still a very insignificant aspect. Wouldn't exactly call it a benefit.

On a related note. What I don't like about the implementation of `AbstractValidatorBase#validate()` is that the exception message is generic. In EDD Bookings, the exception message is sometimes shown in the UI.

For instance on the Bookings page, if an async request comes back as an error, it will contain a `message` that would be the exception message, and the UI will show a toast notification containing that message.

By using `AbstractValidatorBase`, that message will always be ["Problem validating"](https://github.com/Dhii/validation-base/blob/develop/src/AbstractValidatorBase.php#L48).

Reply:
Yes, I agree that the message is annoying. Do you think that the problem can be solved by updating the  `$message` parameter to the [exception factory][] to be `null`? This would result in the first validation reason becoming the exception message.

[exception factory]: https://github.com/Dhii/validation-base/blob/develop/src/CreateValidationFailedExceptionCapableTrait.php#L33

Reply:
Maybe. I was under the impression that the first error is always used. In fact, that's why the 2 checks are in that order. I'll see if your proposed solution works. I always forget what precedence PHP uses for trait and class methods.

Reply:
That was a question for later. The validation base package would need to be updated and released first. Are you going to use the abstract base validator now or not, that is my question.

Reply:
Now. See changes.